packages/cli/src/parallel-tests/parallel-tests.handler.ts

```diff
@@ -303,3 +292,39 @@ const logRetrySummary = (
 
 const testName = (testCase: RerunnableTestCase) =>
   testCase.title != null ? `'${testCase.title}'` : `#${testCase.id + 1}`;
+
+const getNewMergedResult = (
+  testCase: TestCase,
+  currentMergedResult: DetailedTestCaseResult | null,
+  newResult: DetailedTestCaseResult | null
+): DetailedTestCaseResult => {
+  // If currentMergedResult is null then this is our first try, our original head replay
+  if (currentMergedResult == null) {
+    if (newResult == null) {
+      // This means our original head replay failed fatally (not just a failed diff, but failed to even run)
+      // In this case we just return it as result = fail, with no screenshot diffs
+      return {
+        ...testCase,
+        headReplayId: "",
+        result: "fail",
+        screenshotDiffResults: [],
+      };
+    } else {
+      // In this case the newResult is our first head replay, our first result,
+      // so lets initialize the mergedResult to this
+      return newResult;
+    }
+  }
+
+  // If currentMergedResult is not null then newResult is a retry, containing comparison screenshots to our original head replay
+  if (newResult == null) {
+    // In this case newResult is a retry of the head replay, but it failed fatally (not just a failed diff, but failed to even run)
+    // So we just ignore this retry
+    return currentMergedResult;
+  } else {
+    return mergeResults({
+      currentResult: currentMergedResult,
+      comparisonToHeadReplay: newResult,
+    });
+  }
+};
```

Comment:
Style: avoid `else` clauses and nested `if` clauses.

```suggestion
  if (currentMergedResult == null && newResult == null) {
    // This means our original head replay failed fatally (not just a failed diff, but failed to even run)
    // In this case we just return it as result = fail, with no screenshot diffs
    return {
      ...testCase,
      headReplayId: "",
      result: "fail",
      screenshotDiffResults: [],
    };
  }
  
  if (currentMergedResult == null) {
    // In this case the newResult is our first head replay, our first result,
    // so lets initialize the mergedResult to this
    return newResult;
  }
  
  // If currentMergedResult is not null then newResult is a retry, containing comparison screenshots to our original head replay
  if (newResult == null) {
    // In this case newResult is a retry of the head replay, but it failed fatally (not just a failed diff, but failed to even run)
    // So we just ignore this retry
    return currentMergedResult;
  }
  
  return mergeResults({
    currentResult: currentMergedResult,
    comparisonToHeadReplay: newResult,
  });
};
```